eppo_core/src/eval/eval_precomputed.rs

```diff
@@ -0,0 +1,326 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use chrono::{DateTime, Utc};
+
+use crate::eval::get_assignment;
+use crate::precomputed::{PrecomputedAssignment, PrecomputedBandit, PrecomputedConfiguration};
+use crate::ufc::{ConfigurationFormat, ValueWire, VariationType};
+use crate::{Configuration, ContextAttributes, Str};
+
+pub fn get_precomputed_configuration(
+    configuration: Option<&Configuration>,
+    subject_key: &Str,
+    subject_attributes: &Arc<ContextAttributes>,
+    actions: &HashMap<Str, ContextAttributes>,
+    now: DateTime<Utc>,
+) -> PrecomputedConfiguration {
+    let Some(configuration) = configuration else {
+        log::warn!(target: "eppo",
+                   subject_key;
+                   "evaluating a flag before Eppo configuration has been fetched");
+        return PrecomputedConfiguration {
+            obfuscated: serde_bool::False,
+            format: ConfigurationFormat::Precomputed,
+            created_at: now,
+            environment: None,
+            flags: HashMap::new(),
+            bandits: HashMap::new(),
+        };
+    };
+
+    let generic_attributes = Arc::new(subject_attributes.to_generic_attributes());
+
+    let flags = configuration
+        .flags
+        .compiled
+        .flags
+        .keys()
+        .filter_map(|flag_key| {
+            get_assignment(
+                Some(configuration),
+                flag_key,
+                &subject_key,
+                &generic_attributes,
+                None,
+                now,
+            )
+            .unwrap_or_else(|err| {
+                log::warn!(
+                    target: "eppo",
+                    subject_key,
+                    flag_key,
+                    err:?;
+                    "Failed to evaluate assignment"
+                );
+                None
+            })
+            .map(|assignment| ((flag_key.clone(), PrecomputedAssignment::from(assignment))))
+        })
+        .collect::<HashMap<_, _>>();
+
+    let bandits = configuration
+        .bandits
+        .as_ref()
+        .map(|bandits| {
+            configuration
+                .flags
+                .compiled
+                .flags
+                .iter()
+                .filter_map(|(flag_key, flag)| {
+                    let flag = flag.as_ref().ok()?;
+
+                    // Skip non-string variations as they can't be bandits.
+                    if flag.variation_type != VariationType::String {
+                        return None;
+                    }
```

Comment:
I know this is true but this optimization could put us into a tricky spot if the code ships in clients and this fact changes. cc @aarsilv is this ever changing?

```suggestion

```